src/DestroyPlugin.groovy

```diff
@@ -0,0 +1,16 @@
+class DestroyPlugin implements TerraformEnvironmentStagePlugin {
+
+    public static void init() {
+        DestroyPlugin plugin = new DestroyPlugin()
+        
+        // Need to set command name to "destroy"
+        ConfirmApplyPlugin.withCommand("destroy")
+        TerraformEnvironmentStage.addPlugin(plugin)
+    }
+
+    @Override
+    public void apply(TerraformEnvironmentStage stage) {
+        stage.withStrategy(new DestroyStrategy())
+    }
+
+}
```

Comment:
What you did above gave me a new idea!! What if...

```
class DestroyPlugin implements TerraformEnvironmentStagePlugin {

    public static void init() {
        DestroyPlugin plugin = new DestroyPlugin()

        // Need to set command name to "destroy"
        ConfirmApplyPlugin.withConfirmationMessage("YOU ARE ABOUT TO DESTORY YOUR ENVIORNMENT")
        TerraformPlanCommand.withDestroy()
        TerraformApplyCommand.withDestroy()
    }
}
```

```
class TerraformPlanCommand {
...
    public static withDestroy() {
        withArgument('-destory')
    }
}
```

```
class TerraformApplyCommand {
...
    public static withDestroy() {
         this.command = 'destroy'
    }
}
```

1. We would no longer need to modify TerraformEnvironmentStage *at all*.  
2. We wouldn't need TerraformDestroyStage.  
3. We wouldn't need a new TerraformDestoryCommand
4. *ALL* existing plugins would "just work"

Reply:
The above could even be implemented without adding new `withDestroys()` methods.  And in fact, not creating these new methods might be the better/more-conservative approach.